src/logger.c

```diff
@@ -3,8 +3,9 @@
  *
  * SPDX-License-Identifier: Apache-2.0
  */
-
+#ifndef _GNU_SOURCE
 #define _GNU_SOURCE		/* See feature_test_macros(7) */
+#endif
```

Comment:
From a quick look, we are not using anything from `_GNU_SOURCE`. Without this, macro both libosdp and c-utils builds fine for me. Can you please check and remove ` _GNU_SOURCE` here and in libosdp?